ide/coqide/FAQ.md

```diff
@@ -0,0 +1,81 @@
+# CoqIDE FAQ #
+
+### What is CoqIDE?
+
+A powerful GTK3-based graphical interface for Coq.
+See https://coq.inria.fr for more information.
+
+### How to enable Emacs keybindings?
+
+Insert the following line in your `gtkrc` file:
+```
+gtk-key-theme-name = "Emacs"
+```
+The location of this file is system-dependent. If you're running
+GNOME, you may use the graphical configuration tools.
+
+See also the µPG mode in CoqIDE preferences and help.
+
+### How to enable antialiased fonts?
+
+Set the `GDK_USE_XFT` variable to 1. This is the default with Gtk >= 2.2.
+If some of your fonts are not available, set `GDK_USE_XFT` to 0.
+
+### How to use those Forall (∀) and Exists (∃) pretty symbols?
+
+Thanks to the Notation features in Coq, you just need to insert these
+lines in your Coq Buffer :
+```
+Notation "∀ x : t, P" := (forall x:t, P) (at level 200, x ident).
+Notation "∃ x : t, P" := (exists x:t, P) (at level 200, x ident).
+```
+Copy-pasting of these lines from this file will not work outside of CoqIDE.
+You need to load a file containing these lines or to enter the `∀`
+using an input method (see next question below).
+To try it, just use `Require Import Utf8` from inside CoqIDE.
+
+To enable these notations automatically, start `coqide` with:
+```
+coqide -ri Utf8
+```
+
+### How to type non-ASCII symbols?
+
+There are several methods, some of which are described below.
+
+#### Type the unicode code point
+
+The hexadecimal unicode code point of ∀ is 2200 and the one of ∃ is 2203.
+[This Wikipedia page](https://en.wikipedia.org/wiki/Unicode_input#Hexadecimal_input)
+explains how to generate a character from its code point in different systems.
+For instance, under Linux, you must press Ctrl-Shift-U, then type the code point,
+then press Enter.
+
+See https://unicode.org for more code points.
+
+#### Use the LaTeX-to-unicode feature of CoqIDE
+
+You can type e.g. `\forall` or `\exists` in your buffer,
+then (while the cursor is at the end of the word)
+select *Tools > LaTeX-to-unicode* (or, equivalently, type Ctrl-Space or
+Cmd-Space).
+
+#### Use an input method editor
+
+Such editors include SCIM and iBus. The latter offers
+a module for LaTeX-like inputting.
+
+### How to customize the shortcuts for menus?
+
+Two solutions are offered:
+- edit `$XDG_CONFIG_HOME/coq/coqide.keys` by hand; or
+- if your system allows it, from CoqIDE, you may select a menu entry
+  and press the desired shortcut.
+
+### What encoding should I use?
+
+The encoding option is related to the way files are saved.
+Keep it as UTF-8 until it becomes important for you to exchange files
```

Comment:
The preference `Edit/Preferences/Files/File charset encoding` is not used anywhere in CoqIDE.  I think CoqIDE is strictly UTF-8 at this point.

Reply:
Actually, the setting works. I’ve just tested it. You have to type something like `WINDOWS-1252`, then create a non-UTF-8 file with a command such as `echo $'\x9C' > file.v` and you will see the character (`œ` with this code).

Reply:
Ah, I misread the code.  But who in their right mind uses an encoding other than UTF-8 these days?

Reply:
Nobody, I hope. But it’s still a common option to have in a text editor.

Reply:
Proper documentation would provide the list of valid encodings, probably through a hyperlink to the authoritative list.

FWIW, UTF-16 is sometimes used for Asian languages such as Japanese, for which it it more compact.  Not sure how widely used it is.

Reply:
Likely the encoding comment / IDE option comes from the time in Europe was common to use 8859-1 for non-ascii characters, then the transition to UTF-8 (in Unix) happened.

Coq only supports UTF-8 encoded files this day, so this option is indeed just not relevant anymore.

Who is using UTF-16 these days is not very relevant, as Coq only supports UTF-8, however note that for example Windows encodes all its unicode strings in UTF-16, so I wouldn't say that UTF-16 is unused these days.

Due to this, the LSP was developed with position encoding to be UTF-16, which has been a source of pain for servers writes since (as most developers prefers UTF-8)

Reply:
Is it worth to keep the paragraph? Coq needs UTF-8 anyways, so the question of passing a `.v` file to an application that does not understand UTF-8 does not seem the job of Coq or CoqIDE??